### ISSUE TITLE
Work tracker for moving ComponentConfig types to staging repos

### ISSUE THREAD

**Issue by stewart-yu** [kubernetes/kubernetes#67233]
Now, we're making great progress with implementing the KEP about moving ComponentConfig types to staging repos [kubernetes/community#2354](https://github.com/kubernetes/community/pull/2354), and  there are many PRs that are in progress in both in parallel and serial. Here's a work tracker for related PRs. There is a follow-up PR to the initial KEP to accept this KEP as `implementable` as well: https://github.com/kubernetes/community/pull/2515

First, create required meta-repos:
 - [x] k8s.io/apimachinery: [https://github.com/kubernetes/kubernetes/pull/66058](https://github.com/kubernetes/kubernetes/pull/66058)
 - [x] k8s.io/apiserver: [https://github.com/kubernetes/kubernetes/pull/66059](https://github.com/kubernetes/kubernetes/pull/66059)

Preparation work for making the code moves possible:

- [x] Rename KubeConfigFile to Kubeconfig in ClientConnectionConfiguration [https://github.com/kubernetes/kubernetes/pull/67149](https://github.com/kubernetes/kubernetes/pull/67149)
- [x] Remove references to `pkg/apis/componentconfig/install` and fix inconsistencies [https://github.com/kubernetes/kubernetes/pull/67090](https://github.com/kubernetes/kubernetes/pull/67090)
- [x] Write manually-created conversion funcs for shared ComponentConfig types [https://github.com/kubernetes/kubernetes/pull/67159](https://github.com/kubernetes/kubernetes/pull/67159)
- [x] Remove defaulting from shared ComponentConfig types [https://github.com/kubernetes/kubernetes/pull/67207](https://github.com/kubernetes/kubernetes/pull/67207)
- [x] Remove references to the structs that have moved to their own packages [https://github.com/kubernetes/kubernetes/pull/66722](https://github.com/kubernetes/kubernetes/pull/66722)
- [x] Refactor the kube-controller-manager ComponentConfig structs [https://github.com/kubernetes/kubernetes/pull/66993](https://github.com/kubernetes/kubernetes/pull/66993)

kube-scheduler
- [x] Moving KubeSchedulerConfiguration from ComponentConfig API types to staging repos [https://github.com/kubernetes/kubernetes/pull/66916](https://github.com/kubernetes/kubernetes/pull/66916)

kube-controller-manager:
- [x] Moving GenericControllerManagerConfiguration from `pkg/apis/componentconfig` to `k8s.io/controller-manager`:  [https://github.com/kubernetes/kubernetes/pull/67015](https://github.com/kubernetes/kubernetes/pull/67015)
   - Decided as a non-goal for this release at least, instead the generic parts will reside in the kcm repo.
- [x] Move kube-controller-manager ComponentConfig external types to `k8s.io/kube-controller-manager`:  [https://github.com/kubernetes/kubernetes/pull/67469](https://github.com/kubernetes/kubernetes/pull/67469)
- [x] Move the CloudControllerManagerConfiguration to an API group in `cmd/`:  https://github.com/kubernetes/kubernetes/pull/68233 (initial PR that was later carried: https://github.com/kubernetes/kubernetes/pull/67689)

kube-proxy

- [x] Move kubeproxy ComponentConfig external types to `k8s.io/kube-proxy`  https://github.com/kubernetes/kubernetes/pull/67688 

kubelet:
 - [x] Move kubelet ComponentConfig external types to `k8s.io/kubelet` [https://github.com/kubernetes/kubernetes/pull/67263](https://github.com/kubernetes/kubernetes/pull/67263)

General cleanup PR after all the changes:
 - [x] Consolidate componentconfig code standards https://github.com/kubernetes/kubernetes/pull/68195

**Comment by Lion-Wei** [comment]
I can help for kube-proxy part. : )
/assign

**Comment by luxas** [comment]
For reference, this KEP has been moved to the `implementable` stage via this PR: https://github.com/kubernetes/community/pull/2515, so we are good to go for all components here :100:

**Comment by dims** [comment]
Related to https://github.com/kubernetes/kubernetes/issues/57618

**Comment by stewart-yu** [comment]
all fixed in `v1.12`, so close it now
/close
